internal/common/validations.go

```diff
@@ -3,7 +3,38 @@
 
 package common
 
+import (
+	"strings"
+
+	"github.com/go-openapi/swag"
+)
+
 // IsAgentCompatible checks if the given agent image is compatible with what the service expects.
 func IsAgentCompatible(expectedImage, agentImage string) bool {
 	return agentImage == expectedImage
 }
+
+var NonIgnorableHostValidations []string = []string{"connected", "has-inventory", "machine-cidr-defined", "hostname-unique", "hostname-valid"}
+var NonIgnorableClusterValidations []string = []string{"api-vips-defined", "ingress-vips-defined", "all-hosts-are-ready-to-install", "sufficient-masters-count", "pull-secret-set", "cluster-preparation-succeeded"}
+
+func MayIgnoreValidation(validationID string, nonIgnorables []string) bool {
+	if validationID == "all" {
+		return true
+	}
+	return !swag.ContainsStrings(nonIgnorables, validationID)
+}
+
+func MayIgnoreValidations(validationIDs []string, nonIgnorables []string) (bool, string) {
+	result := true
+	cantBeIgnored := []string{}
+	for _, validation := range validationIDs {
+		if validation == "all" {
+			return true, ""
+		}
+		if !MayIgnoreValidation(validation, nonIgnorables) {
+			cantBeIgnored = append(cantBeIgnored, validation)
+			result = false
+		}
+	}
+	return result, strings.Join(cantBeIgnored, ",")
```

Comment:
`strings.Join` is for the sole purpose of displaying something to the user, which this function should not be responsible for. It's very confusing to see the join here as there's no obvious reason for it to happen in this context. The caller should perform the join instead, this function should return a simple array

Reply:
Done, join operation moved to the caller and changed return a string array